packages/workspace/src/workspace/hidden_values.ts

```diff
@@ -89,23 +90,30 @@ const splitElementHiddenParts = <T extends Element>(
   )
 
   const hidden = transformElement({
-    element,
+    element: stateElement,
     transformFunc: ({ value, path }) => (isPartOfHiddenPath(path) ? value : undefined),
     strict: true,
   })
   // remove all annotation types from the hidden element so they don't cause merge conflicts
   hidden.annotationTypes = {}
-  return { hidden, visible }
+  // Workaround for transformElement not being able to omit fields from an ObjectType
+  if (isObjectType(hidden) && isObjectType(workspaceElement)) {
+    // filter out fields that were deleted in the workspace
+    hidden.fields = _.pick(hidden.fields, Object.keys(workspaceElement.fields))
```

Comment:
I'm not sure I understand this one - won't it exclude hidden fields from `hidden`?

Reply:
I think this should also include fields with `_hidden = true` and then it will be ok - maybe something like
```suggestion
    hidden.fields = _.pickBy(hidden.fields, (field, name) => workspaceElement.fields[name] !== undefined && !isHidden(field))
```

Reply:
as you said, it is not a real scenario because we don't support hiding field definitions.
you are correct that if we implement that feature without changing this it will break, not sure that means we need to change it now.

Reply:
:+1: added the details on https://salto-io.atlassian.net/browse/SALTO-1086 so if you prefer not to support that now hopefully we'll find it then

Reply:
changed